server/game/stateWatchers/CardsEnteredPlayThisPhaseWatcher.ts

```diff
@@ -0,0 +1,53 @@
+import { StateWatcher } from '../core/stateWatcher/StateWatcher';
+import { StateWatcherName } from '../core/Constants';
+import { StateWatcherRegistrar } from '../core/stateWatcher/StateWatcherRegistrar';
+import Player from '../core/Player';
+import { EnteredPlayCard, PlayableCard } from '../core/card/CardTypes';
+import Game from '../core/Game';
+import { Card } from '../core/card/Card';
+
+export interface EnteredCardEntry {
+    card: EnteredPlayCard,
+    playedBy: Player
+}
+
+export type ICardsEnteredPlayThisPhase = EnteredCardEntry[];
+
+export class CardsEnteredPlayThisPhaseWatcher extends StateWatcher<EnteredCardEntry[]> {
```

Comment:
Can you add a comment here noting that there's a known issue where this currently doesn't work with leaders